runtime/sema/hints.go

```diff
@@ -45,3 +45,20 @@ func (h *ReplacementHint) Hint() string {
 }
 
 func (*ReplacementHint) isHint() {}
+
+// RemovalHint
+
+type RemovalHint struct {
```

Comment:
Curious about potentially making this more specific, could happen later on however! `RedundantCodeRemovalHint`, etc.